backend/src/email/models/email-message-tx.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  Column,
+  DataType,
+  ForeignKey,
+  Model,
+  Table,
+} from 'sequelize-typescript'
+import { User } from '@core/models'
+import { MessageStatus } from '@core/constants'
+
+// identical to CampaignS3ObjectInterface; to tweak and abstract if necessary
+interface AttachmentS3ObjectInterface {
+  key?: string
+  bucket?: string
+  filename?: string
+  temp_filename?: string
+  error?: string
+}
+
+export enum TransactionalEmailMessageStatus {
+  Unsent = 'UNSENT',
+  // AttachmentTooLargeError = 'ATTACHMENT_TOO_LARGE_ERROR', // unused
+  // TooManyAttachmentsError = 'TOO_MANY_ATTACHMENTS_ERROR', // unused
+  // InvalidParamsError = 'INVALID_PARAMS_ERROR', // unused; these are errors caught by Joi
+  InvalidFromAddressError = 'INVALID_FROM_ADDRESS_ERROR',
+  RateLimitError = 'RATE_LIMIT_ERROR',
+  InvalidMessageError = 'INVALID_MESSAGE_ERROR',
+  UnsupportedFileTypeError = 'UNSUPPORTED_FILE_TYPE_ERROR',
+  MaliciousFileError = 'MALICIOUS_FILE_ERROR',
+  BlacklistedRecipientError = 'BLACKLISTED_RECIPIENT_ERROR',
+  Accepted = 'ACCEPTED',
+  Sent = 'SENT',
+  // unsupported for now
+  // Bounced = 'BOUNCED',
+  // Delivered = 'DELIVERED',
+  // Opened = 'OPENED',
+  // Complaint = 'COMPLAINT',
+}
+
+@Table({ tableName: 'email_messages_tx', underscored: true, timestamps: true })
+export class EmailMessageTx extends Model<EmailMessageTx> {
+  @Column({
+    type: DataType.BIGINT,
+    autoIncrement: true,
+    primaryKey: true,
+    allowNull: false,
+  })
+  id!: number
+
+  @ForeignKey(() => User)
+  @Column({ type: DataType.STRING, allowNull: false })
+  userEmail!: string
+
+  @Column({ type: DataType.STRING, allowNull: false })
+  fromName!: string
+
+  @Column({ type: DataType.STRING, allowNull: false })
+  fromAddress!: string
+
+  @Column({ type: DataType.STRING, allowNull: false })
+  recipient!: string
+
+  @Column({ type: DataType.JSON, allowNull: false })
+  params!: Record<string, string>
+
+  @Column({ type: DataType.BOOLEAN, allowNull: false })
+  hasAttachment!: boolean
+
+  // null if hasAttachment is false
+  @Column({ type: DataType.JSON, allowNull: true })
+  attachmentS3Object!: AttachmentS3ObjectInterface | null
+  // @Column(DataType.STRING)
+  // messageId: string | null
+
+  // supported statuses:
+  // 1. null (upon first creation)
+  // 2. ERROR (if error out at any point along the way)
+  // 3. SENDING (if API is successfully called without error on backend)
+  // other statuses requires callback, which will be handled in a later PR
+  @Column({
+    type: DataType.ENUM(...Object.values(MessageStatus)),
+    allowNull: false,
+  })
+  status!: TransactionalEmailMessageStatus
+
+  // will be null upon first creation
+  @Column({ type: DataType.DATE, allowNull: true })
+  sentAt!: Date | null
```

Comment:
maybe should rename as `acceptedAt` to align with `TransactionalEmailMessageStatus`

at the same time, named it `sentAt` because that's what is used in `email_messages` table. I thought the distinction should be:
- Accepted: accepted by our system
- Sent: received a response from downstream system, say, AWS SES

Not sure if this is too fine a distinction